core/ir/aarch64/codec.c

```diff
@@ -1484,6 +1484,58 @@ encode_opnd_imm4(uint enc, int opcode, byte *pc, opnd_t opnd, OUT uint *enc_out)
     return encode_opnd_int(8, 4, false, 0, 0, opnd, enc_out);
 }
 
+/* cmode4_s_sz_msl: Operand for 32 bit elements' shift amount (shifting ones) */
+
+static inline bool
+decode_opnd_cmode4_s_sz_msl(uint enc, int opcode, byte *pc, OUT opnd_t *opnd)
+{
+    /* cmode size shift amounts
+     * 110x  32   8,16
+     * This is an MSL (Modified Shift Left). Unlike an LSL (Logical Shift
+     * Left), this left shift shifts ones instead of zeros into the low order
+     * bits.
+     *
+     * The element size and shift amount are stored as two 32 bit numbers in
+     * sz_shft. This is a workaround until issue #4393 is addressed.
+     */
+    const int cmode4 = extract_uint(enc, 12, 1);
+    const int size = 32;
+    const int shift = ((cmode4 == 0) ? 8 : 16) | (1 << 28);
+    uint64 sz_shft = ((uint64)size << 32) | shift;
+    *opnd = opnd_create_immed_int(sz_shft, OPSZ_8);
+    return true;
+}
+
+static inline bool
+encode_opnd_cmode4_s_sz_msl(uint enc, int opcode, byte *pc, opnd_t opnd,
+                            OUT uint *enc_out)
+{
+    if (!opnd_is_immed_int(opnd))
+        return false;
+
+    int64 sz_shft = opnd_get_immed_int(opnd);
+    int shift = (int)(sz_shft & 0xffffffff);
+    if (!TEST(1U << 28, shift)) // MSL bit should be set
+        return false;
+    shift &= 0xff;
```

Comment:
Is this just removing the MSL bit?  Clearer to use &= ~CMODE4_MSL_BIT

Reply:
This is just picking out the shift amount from the lowest byte. Seems to be the simplest way.

Reply:
But it's not in the lowest byte: it's the bottom 32 bits, with the top 32 bits being the size, according to the comment above.

Reply:
It seems there is complex structure here: it's not just two 32-bit values then; it seems to be a one-byte value, a special bit, and a size.  Maybe a struct should be defined to avoid all this bit manipulation.